Utils/github_workflow_scripts/utils.py

```diff
@@ -92,25 +92,55 @@ class Checkout:  # pragma: no cover
     previously current branch.
     """
 
-    def __init__(self, repo: Repo, branch_to_checkout: str):
+    def __init__(self, repo: Repo, branch_to_checkout: str, fork_owner: Optional[str] = None, repo_name: str = 'content'):
         """Initializes instance attributes.
         Arguments:
             repo: git repo object
             branch_to_checkout: The branch or commit hash to check out.
+            fork_owner (str): The owner of the forked repository.
+                Leave it as None if the branch is in the same repository.
+            repo_name (str): the name of the forked repo (without the owner)
         """
         self.repo = repo
-        self.repo.remote().fetch(branch_to_checkout)
-        self._branch_to_checkout = branch_to_checkout
+
+        if fork_owner:
+            forked_remote_name = f'{fork_owner}_{repo_name}_{branch_to_checkout}_remote'
+            url = f"https://github.com/{fork_owner}/{repo_name}"
+            try:
+                self.repo.create_remote(name=forked_remote_name, url=url)
+            except Exception as error:
+                print(f'could not create remote from {url}, {error=}')
+                # handle the case where the name of the forked repo is not content
+                if github_event_path := os.getenv("GITHUB_EVENT_PATH"):
+                    try:
+                        payload = json.loads(github_event_path)
+                    except ValueError:
+                        print(f'failed to load GITHUB_EVENT_PATH')
+                        raise ValueError(f'cannot checkout to the forked branch {branch_to_checkout} of the owner {fork_owner}')
+                    # forked repo name includes fork_owner + repo name, for example foo/content.
+                    forked_repo_name = payload.get("pull_request", {}).get("head", {}).get("repo", {}).get("full_name")
+                    self.repo.create_remote(name=forked_remote_name, url=f"https://github.com/{forked_repo_name}")
+                else:
+                    raise
+
+            forked_remote = self.repo.remote(forked_remote_name)
+            forked_remote.fetch(branch_to_checkout)
+            self.branch_to_checkout = f'refs/remotes/{forked_remote_name}/{branch_to_checkout}'
+        else:
+            self.branch_to_checkout = branch_to_checkout
+            self.repo.remote().fetch(branch_to_checkout)
+
         try:
             self._original_branch = self.repo.active_branch.name
         except TypeError:
             self._original_branch = self.repo.git.rev_parse('HEAD')
 
     def __enter__(self):
         """Checks out the given branch"""
-        self.repo.git.checkout(self._branch_to_checkout)
-        return self
```

Comment:
Return self for being able to do: 
``` with Checkout() as c:```

Reply:
True, I guess i added self by mistake, however i do not see any harm in adding it, you can still use it `with Checkout()`.

 in addition you deleted the 
`self.repo.git.checkout(self._branch_to_checkout)` by mistake in the suggestion which i believe you didn't mean it.

Reply:
I think the `return self` is for `with Checkout() as c:` "as c" part.

Reply:
i understood what you meant